### PR TITLE
feat: Add Quick Order missing vocalization (#15233)

### LINKED ISSUE #15233: Quick Order local messages missing vocalization
**Describe the bug**

- It seems some messages on the quick order page are not contained in the global message tag. These local messages don't vocalize with any screen reader, perhaps the new assistive directive can help for this issue.

**Expected behaviour**

- All local messages vocalize for screen readers.

Screen Shot 2022-02-22 at 1 33 44 PM
Screen Shot 2022-02-22 at 1 35 37 PM

### PR DESCRIPTION
**Pull request by dydome**
close #15233

### CODE REVIEW

**Review by developpeurweb — CHANGES_REQUESTED** [review]
Notifications seem to work fine in all SR, **except** the one about items being deleted, mainly because it focuses on the "undo" button. So, either
1. focus the whole alert to include both, message and button, in the narration
... or
2. remove the focus management jump from the button (and just extend the time for the user to be able to get there with the keyboard).

**Review by developpeurweb — DISMISSED** [review]
This works well for JAWS*/NVDA/VO
*Works well on JAWS+Firefox. Bugs with JAWS+Chrome (it will be addressed on a separate bug ticket if persistent).

File: feature-libs/cart/quick-order/components/quick-order/quick-order.component.html [hunk 1]
```diff
@@ -1,136 +1,165 @@
...
+                | cxTranslate
+                  : {
+                      name: error.entry.product.name,
+                      code: error.entry.product.code
+                    }
+            }}
+          </span>
+        </li>
+      </ul>
+      <div aria-live="polite" class="cx-visually-hidden">
```
Comment (Matejk00) [inline feature-libs/cart/quick-order/components/quick-order/quick-order.component.html]: I think this `aria-live="polite"` attribute is not necessary here, as we have all messages wrapped into another aria-live attribute.

File: feature-libs/cart/quick-order/components/quick-order/quick-order.component.html [hunk 2]
```diff
@@ -1,136 +1,165 @@
...
+                  : {
+                      name: warning.entry.product.name,
+                      code: warning.entry.product.code,
+                      quantityAdded: warning.quantityAdded
+                    }
+            }}
+          </span>
+        </li>
+      </ul>
+      <div aria-live="polite" class="cx-visually-hidden">
```
Comment (Matejk00) [inline feature-libs/cart/quick-order/components/quick-order/quick-order.component.html]: I think this `aria-live="polite"` attribute is not necessary here, as we have all messages wrapped into another aria-live attribute.

File: feature-libs/cart/quick-order/components/quick-order/quick-order.component.html [hunk 3]
```diff
@@ -1,136 +1,165 @@
...
-                }
-        }}</span>
-      </li>
-    </ul>
-  </cx-message>
-</ng-container>
+          }}</span>
+        </li>
+      </ul>
+      <div aria-live="polite" class="cx-visually-hidden">
```
Comment (Matejk00) [inline feature-libs/cart/quick-order/components/quick-order/quick-order.component.html]: I think this `aria-live="polite"` attribute is not necessary here, as we have all messages wrapped into another aria-live attribute.

File: feature-libs/cart/quick-order/root/config/quick-order-config.ts [hunk 4]
```diff
@@ -12,6 +12,9 @@ export abstract class QuickOrderConfig {
       maxProducts: number;
       minCharactersBeforeRequest: number;
     };
+    list?: {
+      hardDeleteTimeout: number;
```
Comment (Matejk00) [inline feature-libs/cart/quick-order/root/config/quick-order-config.ts:16]: Can we add any jsdoc what this `hardDeleteTimeout` property stand for?

File: projects/storefrontlib/cms-components/misc/message/message.component.ts [hunk 5]
```diff
@@ -44,16 +47,17 @@ export class MessageComponent {
...
-      case GlobalMessageType.MSG_TYPE_CONFIRMATION:
-        return ICON_TYPE.SUCCESS;
       case GlobalMessageType.MSG_TYPE_WARNING:
         return ICON_TYPE.WARNING;
       case GlobalMessageType.MSG_TYPE_ERROR:
         return ICON_TYPE.ERROR;
       case GlobalMessageType.MSG_TYPE_INFO:
         return ICON_TYPE.INFO;
+      default:
+        // GlobalMessageType.MSG_TYPE_CONFIRMATION:
```
Comment (Matejk00) [inline projects/storefrontlib/cms-components/misc/message/message.component.ts]: Should we have commented this out?

**Review by Matejk00 — CHANGES_REQUESTED** [review]

**Review by Matejk00 — APPROVED** [review]

**Pull request merged**